# HG changeset patch
# User Johannes Zarl <johannes.zarl@jku.at>
# Date 1410483147 14400
#      Thu Sep 11 20:52:27 2014 -0400
# Node ID 96df707c7dadc1bb8c69cb34e71b76b6a032ad32
# Parent  d59d052c9858d534e26b472f82bbf191c036f5eb
Do not include system header files in extern "C" block (bug #43171)

* quit.h: Include system header files (which may be provided by gnulib
and may have C++-isms) outside of the extern "C" block.

diff --git a/liboctave/cruft/misc/quit.h b/liboctave/cruft/misc/quit.h
--- a/liboctave/cruft/misc/quit.h
+++ b/liboctave/cruft/misc/quit.h
@@ -18,26 +18,26 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_quit_h)
 #define octave_quit_h 1
 
+#include <stdio.h>
+
+#include <signal.h>
+#include <setjmp.h>
+
 #ifdef __cplusplus
 #include <new>
 extern "C" {
 #endif
 
-#include <stdio.h>
-
-#include <signal.h>
-#include <setjmp.h>
-
 #if defined (__WIN32__) && ! defined (_POSIX_VERSION)
 
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 
 CRUFT_API extern void w32_sigint_init (void);   /* setup */
 CRUFT_API extern void w32_raise_final (void);   /* tear down */
 CRUFT_API extern void w32_raise (int sig);      /* raise signal in main thread */
